### PR TITLE
chore: extend danger to ensure major updates for all paste packages have upgrade guide

### PR DESCRIPTION
**Pull request by jb-twilio**
This PR expands an existing DangerJS rule. Previously we intended to check if `major` updates to `@twilio-paste/core` included a corresponding upgrade guide change to our doc site. Now we decidedly check against **any** package `major` update. There could be exceptions here around private packages that might be versioned for whatever reason? In those cases I'm all ears to adjust further.

Integration test PRs:

https://github.com/twilio-labs/paste/pull/2775 (should fail DangerJS - major updates across multiple packages **without** upgrade guide)
https://github.com/twilio-labs/paste/pull/2777 (should pass DangerJS - major updates across multiple packages **with** upgrade guide)
https://github.com/twilio-labs/paste/pull/2772 (should fail DangerJS- major updates to core package **without** upgrade guide)
https://github.com/twilio-labs/paste/pull/2776 (should pass DangerJS - major updates to core package **with** upgrade guide)
https://github.com/twilio-labs/paste/pull/2773 (should pass DangerJS - minor and patch updates **without** upgrade guide)

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: .danger/major-missing-upgrade-guide.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export const FAIL_MESSAGE = `You have marked a Paste package as a MAJOR update via changeset, but have not provided a corresponding [upgrade guide](https://paste.twilio.design/core/upgrade-guide) entry.
+Please update the guide to inform our users what to do in response to this major update to Paste.`;
+
+/**
+ * Check if the changeset file contents include a mention of a package major update
+ *
+ * @param {string} filePath
+ * @return {*}
+ */
+export const hasMajorUpdate = (filePath: string) => {
```
Comment (nkrantz) [inline .danger/major-missing-upgrade-guide.ts:21]: suggestion: we are now using both "upgrade" and "update" here. Is there a reason you changed it from "hasMajorUpgrade" to "hasMajorUpdate"? curious to know what the distinction is, and if there's no difference, I'd recommend we use consistent language for clarity

**Review by nkrantz — COMMENTED** [review]
Thanks for adding this important check, Jeff!

**Review by TheSisb — APPROVED** [review]
Good improvements!

File: .danger/major-missing-upgrade-guide.ts (hunk 1 above)
Comment (nkrantz) [reply .danger/major-missing-upgrade-guide.ts:21]: ok, thanks for the clarification. makes sense!

**Review by nkrantz — APPROVED** [review]

**Pull request merged**